### PR TITLE
Iris multiview/aic 5764

### PR DESCRIPTION
**Pull request by ShevchenkoRostyslav**
# Add Template Alignment and Identity Validation Nodes

This PR introduces two new nodes to the multi-frame IRIS Pipeline.

### 1. Hamming Distance-Based Template Alignment (`HammingDistanceBasedAlignment`)
- **Purpose**: Aligns iris templates by finding optimal rotation offsets to minimize Hamming distances
- **Key Features**:
  - Automatic reference template selection using multiple aggregation methods (linear, mean squared, root mean squared)
  - Configurable rotation shift limits (default: 15)
  - Returns both aligned templates and original pairwise distances
  - Supports normalized and non-normalized Hamming distance calculations

```[yaml]
- name: templates_alignment
      algorithm:
        class_name: iris.nodes.templates_alignment.hamming_distance_based.HammingDistanceBasedAlignment
        params:
          rotation_shift: 15
          use_first_as_reference: false
          normalise: true
          reference_selection_method: linear
      inputs:
        - name: templates
          source_node: input
      callbacks:
```

### 2. Template Identity Filter (`TemplateIdentityFilter`)
- **Purpose**: Ensures all templates are from the same identity
- **Key Features**:
  - Uses Breadth-First Search and pairwise Hamming distances to detect identity clusters
  - If more than 1 identity has been detected - raise an alarm
  - Configurable validation actions: remove outliers, raise errors, or log warnings
  - Minimum template count validation after filtering
  - Can work with precomputed distances from alignment node

An example of the clustering search algorithm:
```
Suppose you have 6 templates (0 to 5) and a distances dict like:
{
  (0, 1): 0.1,
  (1, 2): 0.15,
  (0, 2): 0.2,
  (3, 4): 0.05,
  (4, 5): 0.08,
  (3, 5): 0.2,
}
With threshold = 0.15 and min_cluster_size = 2, the function would return 2 clusters:
[{0, 1, 2}, {3, 4, 5}]
```

### Multiframe Aggregation Pipeline Updates
- Added `templates_alignment` node before aggregation
- Added `identity_validation` node between alignment and aggregation
- Updated `templates_aggregation` to use filtered templates
- Adjusted mask threshold from 0.5 to 0.01 for better aggregation
- Updated pipeline version to 1.7.2

### Configuration Changes
- Pipeline version bumped from 1.7.1 to 1.7.2
- Templates aggregation sub-pipeline version updated to 1.7.1
- Added comprehensive configuration for both new nodes

## Error Handling

- Added `IdentityValidationError` exception class for identity validation failures
- Robust error handling for edge cases (no clusters, multiple clusters, insufficient templates)

### Unit Tests
- **Template Alignment**: 15 comprehensive test cases covering rotation, reference selection methods, and edge cases
- **Identity Filter**: 20+ test cases covering clustering, purification, and validation scenarios
- **Integration**: End-to-end tests for both nodes in pipeline context

### Test Coverage
- Reference selection method comparisons and numerical validation
- Edge case handling (empty inputs, single templates, tie-breaking)
- Different aggregation method sensitivities to outliers
- Pipeline integration testing

## Type

- [x] Feature
- [ ] Refactoring
- [ ] Bugfix
- [ ] DevOps
- [ ] Testing

## Checklist

- [x] I've made sure that my code works as expected by writing unit tests.
- [x] I've checked if my code doesn't generate warnings or errors.
- [x] I've performed a self-review of my code.
- [x] I've made sure that my code follows the style guidelines of the project.
- [x] I've commented hard-to-understand parts of my code.
- [x] I've made appropriate changes in the documentation.

### CODE REVIEW

File: tests/unit_tests/nodes/templates_filter/test_single_identity_filter.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+def test_find_identity_clusters_two_clusters():
+    mat = np.array([[0, 0.05, 0.8, 0.8], [0.05, 0, 0.8, 0.8], [0.8, 0.8, 0, 0.05], [0.8, 0.8, 0.05, 0]])
+    d = make_dist_dict(mat)
+    clusters = find_identity_clusters(d, n=4, threshold=0.2)
+    assert len(clusters) == 2
+    assert {0, 1} in clusters and {2, 3} in clusters
+
+
+def test_find_identity_clusters_clusters_with_outliers():
```
Comment (TanguyJeanneau) [inline tests/unit_tests/nodes/templates_filter/test_single_identity_filter.py:102]: me see many relevant unit tests.

me happy 🌻

Comment (TanguyJeanneau) [inline tests/unit_tests/nodes/templates_filter/test_single_identity_filter.py:102]: me see many relevant unit tests.

me happy 🌻

**Review by TanguyJeanneau — COMMENTED** [review]
Nice feature, well implemented ! 🙌 
Most of my comments re: 1`HammingDistanceBasedAlignment` were resolved in #109, I've left a few minor re: `TemplateIdentityFilter`

**Review by TanguyJeanneau — APPROVED** [review]
LGTM ! 🙌
